src/main/java/org/springframework/hateoas/UriTemplate.java

```diff
@@ -67,6 +67,7 @@ public UriTemplate(String template) {
 			String[] names = matcher.group(2).split(",");
 
 			for (String name : names) {
+				name = name.endsWith("*")? name.substring(0, name.length() - 1) : name;
```

Comment:
The bit of the pattern that is relevant can be found here:
https://tools.ietf.org/html/rfc6570#section-2.4.2
We have been running with a patched version of the library since August with this fix in it.